### DISCUSSION TITLE
Mixing raylib components and raygui controls in a single window instance.

### DISCUSSION THREAD

**Discussion by Axle-Ozz-i-sofT** [raysan5/raylib#2497] [General]
Hi Ray

I have been searching around in an attempt to confirm if it is possible to mix raylib and raygui. I have encountered some ambiguity in my search for a conclusive answer to this.

An example would be a simple paint like drawing application with a draw pad using the raylib API and common controls from the raygui (such as buttons (Save-Open-Clear), colour picker control),

My first "assumption" was that I would be able to mix the 2 APIs, but have seen suggestions that I can't do this.

Q1. Can I mix raylib and raygui components in a single window instance?
Q2. If yes, would It be best to give focus to raylib or raygui? (Both appear to use InitWindow() so I would guess it wouldn't matter)
Q3. I read that raylib/gui does not support child windows (As a second window instance, not MDI style). Is this correct?
      (It is C so ultimately I should be able to start a second InitWindow() process).

I know I am preempting some questions before testing them in the IDEs, but I need to get a reasonable picture before I start as I will be coding all examples I do in C, FreeBASIC and Python3 at the same time on Windows and Ubuntu and have to set myself up to make my life as easy as I can. Writing books with step by step and pictures as well creating all of the example in each environment is a time consuming and taxing exercise :)
( I still have to keep up with daily life commitments as well as my tertiary submissions all while dealing with chronic heart failure. Life wasn't meant to be easy someone said lol)

Best regards
Axle

P.S. I did see some examples at https://github.com/Demizdor/ so I am guessing I can mix drawing and GUI components.

**Comment by raysan5** [comment]
Not sure if I understand the issue. Of course you can mix raylib and raygui drawing, actually raygui functions just call `DrawRectangle()`/`DrawText()` functions.

 > Q1. Can I mix raylib and raygui components in a single window instance?

Yes.

 > Q2. If yes, would It be best to give focus to raylib or raygui? (Both appear to use InitWindow() so I would guess it wouldn't matter)

I don't understand the question. There is no "focus" concept, you just draw things where you want in the screen.

 > Q3. I read that raylib/gui does not support child windows (As a second window instance, not MDI style). Is this correct?

Yes, raylib does not support child window, neither raygui (because raygui uses raylib and raylib does not support child windows). In any case, you can customize raylib to support child windows if desired, it's [explained in the Wiki](https://github.com/raysan5/raylib/wiki/Use-multiple-windows).

**Reply by Axle-Ozz-i-sofT** [reply]
Thank you Ray
You answered my questions successfully (As ambiguous as they may have appeared lol).

Not sure how I missed the wiki section, Or I read over it and then forgot. Sorry for that.

Axle
